nautobot/docs/development/core/kubernetes-job-support.md

```diff
@@ -0,0 +1,509 @@
+media-root        Bound    pvc-011f484b-2ccf-4fe0-953b-289a13ad0480   200Mi      RWO            standard       <unset>                 2m35s
+pgdata-nautobot   Bound    pvc-5954eb3f-75e3-4f6b-9b9c-a91e40ea96bf   200Mi      RWO            standard       <unset>                 2m35s
+```
+
+To confirm all required kubernetes entities are up and running, run the following command:
+
+```bash
+kubectl get all
+```
+
+You should see the following output, note that pods are automatically created when we create deployments:
+
+```bash
+NAME                               READY   STATUS    RESTARTS      AGE
+pod/celery-beat-6fb67477b7-rsw62   1/1     Running   0             30m
+pod/db-8687b48964-gtvtc            1/1     Running   0             30m
+pod/nautobot-679bdc765-pl2ld       1/1     Running   0             30m
+pod/redis-7cc58577c-tl5sq          1/1     Running   0             30m
+
+NAME                 TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+service/db           ClusterIP   10.111.0.30     <none>        5432/TCP   30m
+service/kubernetes   ClusterIP   10.96.0.1       <none>        443/TCP    30m
+service/nautobot     ClusterIP   10.106.32.53    <none>        8080/TCP   30m
+service/redis        ClusterIP   10.102.99.143   <none>        6379/TCP   30m
+
+NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/celery-beat   1/1     1            1           30m
+deployment.apps/db            1/1     1            1           30m
+deployment.apps/nautobot      1/1     1            1           30m
+deployment.apps/redis         1/1     1            1           30m
+```
+
+## Environment Variables and Job Pod Manifest
+
+## Running Jobs
+
+To pass stdin into the nautobot container that exists with in the nautobot pod, first you need to get the name of the nautobot pod.
+
+You can execute the following command to see all the pods.
+
+```bash
+kubectl get pods
+```
+
+You should see the following output and note that the pod with the prefix `nautobot-*`. That is what we need for our command.
+
+```bash
+NAME                               READY   STATUS    RESTARTS      AGE
+pod/celery-beat-6fb67477b7-rsw62   1/1     Running   0             30m
+pod/db-8687b48964-gtvtc            1/1     Running   0             30m
+pod/nautobot-679bdc765-pl2ld       1/1     Running   0             30m
+pod/redis-7cc58577c-tl5sq          1/1     Running   0             30m
+```
+
+Now to get to the CLI of the nautobot container in the pod, we execute the following command:
+
+```bash
+kubectl exec --stdin --tty <nautobot-pod-name> -- /bin/bash
+```
+
+You should see that we are inside the container as the root user:
+
+```bash
+root@nautobot-679bdc765-pl2ld:/source#
+```
+
+### Create and Assign a Kubernetes Job Queue to a Job
+
+Now we are inside the container, we can access `nbshell` using `nautobot-server nbshell` and make modifications to our database records. You should see the following output:
```

Comment:
Do we not have the ability to connect to the k8s deployment with a web browser rather than needing to do all the below through nbshell? I would have thought that should "just work" given the nautobot pod is publishing port 8080.